File: ios/AgentControlCenter/Models/Agent.swift
import Foundation

// MARK: - Agent Models
// 对应 protocol/schemas/agent-schema.json

/// Agent 类型枚举，决定 TransportFactory 路由到具体传输实现。
enum AgentType: String, Codable, CaseIterable {
    case hermes = "Hermes"
    case openCode = "OpenCode"
    case openClaw = "OpenClaw"
    case openAI = "OpenAI"
    case xiaomiMiMo = "XiaomiMiMo"
    case localModel = "LocalModel"
    case comfyUI = "ComfyUI"
    case openWebUI = "OpenWebUI"

    var displayName: String { rawValue }
}

/// Agent 通信协议类型
enum AgentProtocol: String, Codable {
    case webSocket = "WebSocket"
    case httpSSE = "HttpSSE"
    case mcp = "MCP"
    case local = "Local"
}

/// Agent 连接状态
enum AgentStatus: String, Codable {
    case online = "Online"
    case offline = "Offline"
    case connecting = "Connecting"
    case error = "Error"
}

/// Agent 能力声明
enum AgentCapability: String, Codable, CaseIterable {
    case chat = "CHAT"
    case task = "TASK"
    case workflow = "WORKFLOW"
    case mcp = "MCP"
    case filesystem = "FILESYSTEM"
    case terminal = "TERMINAL"
    case voice = "VOICE"
    case imageGen = "IMAGE_GEN"
    case codeExecution = "CODE_EXECUTION"
}

/// Agent 连接配置。apiKey 在持久化时由 Keychain 加密，使用 AKS: 前缀格式。
struct AgentConfig: Codable, Identifiable, Equatable, Sendable {
    var id: String = "default"
    var name: String = "Default Agent"
    var type: AgentType = .hermes
    var serverUrl: String = ""
    var apiKey: String = ""
    var model: String = ""
    var systemPrompt: String = ""
    var temperature: Float = 0.7
    var maxTokens: Int = 4096
    /// 通信协议类型（AgentProtocol.rawValue：WebSocket / HttpSSE / MCP / Local）
    var protocolType: AgentProtocol = .webSocket
}

/// Agent 运行时实例
struct Agent: Codable, Identifiable, Equatable, Sendable {
    var id: String
    var name: String
    var endpoint: String = ""
    var status: AgentStatus = .offline
    var capabilities: [AgentCapability] = [.chat]
    var protocolType: AgentProtocol = .webSocket
    var config: AgentConfig? = nil

    enum CodingKeys: String, CodingKey {
        case id, name, endpoint, status, capabilities, config
        case protocolType = "protocol"
    }
}

File: android/app/src/main/java/com/agentcontrolcenter/app/agent/model/Agent.kt
package com.agentcontrolcenter.app.agent.model

/**
 * Agent — Agent Control Center 的第一公民。
 *
 * 统一模型：不再区分 OpenCode / Claude / OpenAI / Ollama，
 * 只看 [capabilities]。UI 和 Runtime 均通过能力集调度，
 * 而非硬编码 Agent 类型。
 *
 * 定位：Agent Control Center 是统一连接、管理和调度本地与远程 AI Agent 的移动控制中心。
 *
 * @see AgentCapability
 * @see AgentStatus
 */
data class Agent(
    val id: String,
    val name: String,
    /** 连接端点（URL 或 host:port）。 */
    val endpoint: String = "",
    /** 当前连接状态。 */
    val status: AgentStatus = AgentStatus.Offline,
    /** 该 Agent 支持的能力集；UI 据此决定展示哪些功能入口。 */
    val capabilities: List<AgentCapability> = listOf(AgentCapability.CHAT),
    /** 传输协议类型，决定走 WebSocket / HTTP+SSE / MCP 等。 */
    val protocol: AgentProtocol = AgentProtocol.WebSocket,
    /** 关联的 AgentConfig（向后兼容，迁移期保留）。 */
    val config: AgentConfig? = null
)

/**
 * Agent 传输协议类型。定义已迁移至 [AgentProtocol.kt]（含 rawValue / fromRawValue，
 * 与 iOS AgentProtocol 跨端对齐）。此处不再重复声明，避免 Redeclaration 冲突。
 */

/**
 * Agent 当前状态。
 */
enum class AgentStatus(val displayName: String) {
    Online("Online"),
    Offline("Offline"),
    Connecting("Connecting"),
    Error("Error")
}

/**
 * Agent 能力集 — UI 不再判断 Agent 类型，只看能力。
 *
 * 例如：
 * - MacBook 上的 OpenCode → [CHAT, TASK, TERMINAL, FILESYSTEM]
 * - 服务器上的 OpenManus  → [CHAT, TASK, WORKFLOW]
 * - NAS 上的 Ollama       → [CHAT]
 * - Claude Code           → [CHAT, TASK, MCP]
 */
enum class AgentCapability(val displayName: String) {
    CHAT("Chat"),
    TASK("Task Execution"),
    WORKFLOW("Workflow"),
    MCP("MCP Support"),
    FILESYSTEM("Filesystem"),
    TERMINAL("Terminal"),
    VOICE("Voice"),
    IMAGE_GEN("Image Generation"),
    CODE_EXECUTION("Code Execution")
}
